### PR TITLE
Vaults assignment: Todd Li

### PR DESCRIPTION
**Pull request by forcey**

### CODE REVIEW

File: contracts/Vault1.sol [hunk 1]
```diff
@@ -1,6 +1,23 @@
 //SPDX-License-Identifier: Unlicense
 pragma solidity ^0.8.4;
 
-contract Vault1 {
-    
+import "@openzeppelin/contracts/token/ERC20/extensions/ERC20Wrapper.sol";
+
+contract Vault1 is ERC20Wrapper {
```
Comment (shubham-kanodia) [inline contracts/Vault1.sol:6]: Inheriting from ERC20Wrapper is not required, here you are creating a new token (Vault1) whereas the contract should be able to accept any ERC20 token. The best way to do that would be to take the ERC20 address in constructor and then create a token object

File: contracts/Vault2.sol [hunk 2]
```diff
@@ -1,6 +1,26 @@
...
+    event Burned(address owner, uint amount);
+
+    constructor() 
+        ERC20("Vault2", "VALT2") {}
+
+    function mint(uint _amount) public payable {
+        require(_amount != 0, "Must mint a non-zero value.");
+        require(msg.value == _amount, "Must send exactly the same amount of ether.");
+        _mint(msg.sender, _amount);
+        emit Minted(msg.sender, _amount);
```
Comment (shubham-kanodia) [inline contracts/Vault2.sol:17]: _mint already emits an event indicating a `mint` - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L267. `Minted` event can be removed

File: contracts/Vault2.sol [hunk 3]
```diff
@@ -1,6 +1,26 @@
...
+    }
+
+    function burn(uint _amount) public {
+        require(_amount != 0, "Must burn a non-zero value.");
+        _burn(msg.sender, _amount);
+        payable(msg.sender).transfer(_amount);
+        emit Burned(msg.sender, _amount);
```
Comment (shubham-kanodia) [inline contracts/Vault2.sol:24]: _burn already emits an event indicating a `burn` - https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol#L295

**Comment by forcey** [comment]
Thanks Shubham!

**Pull request closed without merging**